README.md

```diff
@@ -5,757 +5,886 @@
-    * [unshift(value: any) => microstate](#unshift-any--value1-any--microstate)
-    * [filter(fn: value => boolean) => microstate](#filterfn-value--boolean--microstate)
-    * [map(fn: (microstate, index) => any) => microstate](#mapfn-microstate-index--any--microstate)
-    * [clear() => microstate](#clear--microstate)
-  * [`Object`](#object)
-    * [set(value: any) => microstate](#setvalue-any--microstate-4)
-    * [assign(object: Object) => microstate](#assignobject-object--microstate)
-    * [put(key: string, value: any) => microstate]()
-    * [delete(key: string) => microstate]()
+  * [Primitive type transitions](#primitive-type-transitions)
+  * [class type transitions](#class-type-transitions)
+  * [chaining transitions](#chaining-transitions)
+  * [initialize transition](#initialize-transition)
+  * [set transition](#set-transition)
+  * [Scope rules](#scope-rules)
+* [microstates npm package](#microstates-npm-package)
+  * [create(Type, value): Microstate](#createtype-value-microstate)
+  * [from(any): Microstate](#fromany-microstate)
+  * [map(fn: tree =&gt; tree, microstate: Microstate): Microstate](#mapfn-tree--tree-microstate-microstate-microstate)
+* [Middleware](#middleware)
+* [Observable Microstates](#observable-microstates)
+* [The Vision of Microstates](#the-vision-of-microstates)
+  * [Shared Solutions](#shared-solutions)
+  * [Added Flexibility](#added-flexibility)
+  * [Framework Agnostic Solutions](#framework-agnostic-solutions)
 * [FAQ](#faq)
   * [What if I can't use class syntax?](#what-if-i-cant-use-class-syntax)
   * [What if I can't use Class Properties?](#what-if-i-cant-use-class-properties)
 * [Run Tests](#run-tests)
 
-## Intro
+<!-- tocstop -->
+</details>
 
-Microstates are a _typed_, _composable_ and _immutable_ state container.
+# Features
 
-Let's get some terminology out of the way so we're on the same page for the rest of the README.
+With microstates added to your project, you get:
 
-* _Typed_ means that microstates know about the kind of data it contains and transitions that can be
-  performed on that data.
-* _Composable_ means that simple types can be grouped to describe complex data structures.
-  Microstates knows how to transition complex data structures in an immutable way.
-* _Immutable_ means that state in the microstate cannot be modified. To change the state, you must invoke a transition.
-* _Transition_ is an operation that has access to the current state, might perform computations/transformations on it and returns a new state.
+* 🍇 Composable type system
+* 🍱 Reusable state atoms
+* 💎 Pure immutable state transitions without writing reducers
+* ⚡️ Lazy and synchronous out of the box
+* 🦋 Most elegant way to express state machines
+* 🎯 Transpilation free type system
+* 🔭 Optional integration with Observables
+* ⚛ Use in Node.js and browser
 
-To add microstates to your project,
+But, most imporantly, Microstates makes working with state fun.
```

Comment:
we need to settle on a representation for Microstates that is consistent throughout the document.  above we use `microstates` but _mostly_ it seems to trend towards the proper nounified `Microstates`, which I'd prefer if given a choice between the two.  `microstate` does not seem to be a generic noun outside of the realms of physics and politics, so it seems worthy of the designation.  but i'm less hung up on which so much as that we're consistent throughout.

Reply:
@sadatay I've been using `Microstates` when I refer to the library or the tool and `microstate` when I'm referring to an instance of the `Microstate` class